packages/knip/src/plugins/msw/index.ts

```diff
@@ -0,0 +1,34 @@
+import { hasDependency } from '../../util/plugin.js';
+import { toEntryPattern, toProductionEntryPattern } from '../../util/protocols.js';
+import type { GenericPluginCallback, IsPluginEnabledCallback } from '../../types/plugins.js';
+
+// https://mswjs.io/docs/integrations/browser
+
+const NAME = 'Mock Service Worker';
+
+const ENABLERS = ['msw'];
+
+const isEnabled: IsPluginEnabledCallback = ({ dependencies }) => hasDependency(dependencies, ENABLERS);
+
+const ENTRY_FILE_PATTERNS = [  '**/mockServiceWorker.{js,ts}',
+  'mocks/browser.{js,ts}',
+  'mocks/handlers.{js,ts}"',
+  'mocks/index.{js,ts}"',
+  'mocks/server.{js,ts}"',
```

Comment:
Are these the default locations?

Feels like files like those listed and `src/mocks/node.js` could be located anywhere without clear defaults, right? If most projects would follow a certain convention that could work too, but I'm not aware of any. Do you have any insights or links that could help here?

Reply:
The default everyone does is in a root directory of /mocks see all the examples - https://github.com/mswjs/examples/tree/main/examples/with-vitest - this is vitest, but its true for all examples.
And here is another example of the /mocks dir - https://mswjs.io/docs/integrations/browser#conditionally-enable-mocking

Reply:
I'm not 100% sure if I'm using the correct knip functions to check for these files. The tests pass and it works - just not sure.

Reply:
Also, if we could peek inside their package.json we could get the correct directory:

```
"msw": {
  "workerDirectory": "public"
}
```

Reply:
Thanks!

The `findReferences` function receives a second `options` arguments which contains `options.manifest`.

Reply:
> I'm not 100% sure if I'm using the correct knip functions to check for these files. The tests pass and it works - just not sure.

Many tests are more "end to end" and use the `main()` function, then you can assert things on the returned `issues` and `counters` objects. See e.g. https://github.com/webpro/knip/blob/main/packages/knip/test/plugins/ava2.test.ts

Reply:
```
const localConfig = basename(configFilePath) === 'package.json' ? manifest.msw : undefined;
const workerDirectory = localConfig?.workerDirectory;

if (workerDirectory) {
  ENTRY_FILE_PATTERNS.push(`${workerDirectory}/mockServiceWorker.{js,ts}`)
}
```

This bit of code looks for the entry somewhere else other than the default of public